tests/unit/array/test_sequence.py

```diff
@@ -103,3 +103,31 @@ def test_context_manager_from_disk(storage, config, start_storage, tmpdir, tmpfi
 
     del da
     del da2
+
+
+@pytest.mark.parametrize('index', [1, '1', slice(1, 2), [1]])
+def test_sqlite_del_and_append(index):
+    da = DocumentArray(storage='sqlite')
+
+    with da:
+        da.extend([Document(id=str(i)) for i in range(5)])
+    with da:
+        del da[1]
+        da._save_offset2ids()
+        da.append(Document(id='new'))
+
+    assert da[:, 'id'] == ['0', '2', '3', '4', 'new']
+
+
+@pytest.mark.parametrize('index', [1, '1', slice(1, 2), [1]])
+def test_sqlite_del_and_append(index):
+    da = DocumentArray(storage='sqlite')
+
+    with da:
+        da.extend([Document(id=str(i)) for i in range(5)])
+    with da:
+        da[1] = Document(id='new')
+        da._save_offset2ids()
```

Comment:
this is an awkward test. Why do we need to test something with the private method used?